### ISSUE TITLE
BitBucket: No linked repo was found for project when using --org

### ISSUE THREAD

**Issue by ataylorme** [pantheon-systems/terminus-build-tools-plugin#290]
the repo is not assigned to any project in Bitbucket and I get that message. This is my command:

```
terminus build:project:create \
--team="My Team" \
--email="developer@myteam.com" \
--admin-email="developer@myteam.com" \
--admin-password="cms-admin-password" \
--git=bitbucket \
--org=MyBitbucketOrg \
--stability=dev \
--visibility=private \
--label="Label project name" \
pantheon-systems/example-wordpress-composer:dev-master \
my-project-name
```

After running the command, it creates the repo in a random project inside my organization in Bitbucket, and the message there is:

`No linked project was found for this repo. Head on over to Rollbar.com to set it up.`

**Comment by ataylorme** [comment]
@nelsonamaya82 I moved this to a new issue so we can chat more without alerting the 20 people following the other thread

[The current code](https://github.com/pantheon-systems/terminus-build-tools-plugin/blob/master/src/ServiceProviders/RepositoryProviders/Bitbucket/BitbucketProvider.php#L51) is using the [`{org}/{repo}/repositories` API endpoint](https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D#repositoryResource-POSTanewrepository)

The repo is being created but no `project` information is passed. According to the API docs we should be able to send `project` info.

Perhaps for BitBucket we need `project` and `org` information. Or would you expect a new project to be created?

**Comment by ataylorme** [comment]
@aaronbauman are you using `--org` with BitBucket? If so, I would be curious how this has been behaving for you.

**Comment by ataylorme** [comment]
If we go with choosing an existing project the details can be retrieved from [the `teams/{username}/projects` API endpoint](https://developer.atlassian.com/bitbucket/api/2/reference/resource/teams/%7Busername%7D/projects/#get)

**Comment by ataylorme** [comment]
> Perhaps for BitBucket we need `project` and `org` information. Or would you expect a new project to be created?

We could allow both if `project` info is optional. If the option is passed, use an existing project, otherwise, create a new project.

**Comment by nelsonamaya82** [comment]
Thanks @ataylorme,

> Perhaps for BitBucket we need project and org information. Or would you expect a new project to be created?

I think we need project and org information. If the project already exists, then assign the repo to that project, else create the project with the assigned name.

**Comment by ataylorme** [comment]
@nelsonamaya82 do you think the existing project name in BitBucket will always match the project name give to `build:project:create`?

I'm wondering if we could re-use the existing project name rather than creating a new `project` option.

So `terminus build:project:create --git=bitbucket wp my-project-name` would look for a BitBucket project `my-project-name` and create one if it does not exist

**Comment by nelsonamaya82** [comment]
@ataylorme I think the project name should be a different option than the repo name. Based on the tests I've done, the "project name" we are passing to the `build:project:create` is not actually the project name but the repo name.

So in my case for example, my project name would be my customer name, and inside that project I could have more than one repo. Then I could create multiple repos in the same project this way:

`terminus build:project:create --git=bitbucket --org="My Organization" --project="My Customer" wp my-repo-1`

`terminus build:project:create --git=bitbucket --org="My Organization" --project="My Customer" wp my-repo-2`

BTW the `No linked project was found for this repo` message only happens when passing an `--org` value to the command. When I created it with my personal BB account instead of my organization account, it just works fine.

**Comment by nelsonamaya82** [comment]
Or maybe we could pass the project name in the same `--org` flag, like `--org:"My Organization/My Project"`.

**Comment by ataylorme** [comment]
> the No linked project was found for this repo message only happens when passing an --org value to the command. When I created it with my personal BB account instead of my organization account, it just works fine.

Yes, I think that is because individual user accounts don't have projects like an organization

**Comment by nelsonamaya82** [comment]
@ataylorme, based on the [{org}/{repo}/repositories API endpoint](https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D), I think the param we need to add to the [$postData](https://github.com/pantheon-systems/terminus-build-tools-plugin/blob/master/src/ServiceProviders/RepositoryProviders/Bitbucket/BitbucketProvider.php#L55) to assign a project is `name`, but I'm not sure how to implement or test it. Also I'm not sure how the API will respond when the project exists or doesn't exist.

**Comment by nelsonamaya82** [comment]
Hi @ataylorme and @aaronbauman, any news on this?  Thanks!

**Comment by aaronbauman** [comment]
Yes we're using `--org` and it's working fine.
The repo that gets created and assigned to a project called "No Project."
(I don't know if this is standard BB behavior, or how this is set up. I can't find any "default project" settings in bb admin.)

imo `--name` is too ambiguous, we should use a `--project` flag to match bitbucket's nomenclature.

I will work on a PR

**Comment by aaronbauman** [comment]
Looking into this a bit further.
In Gitlab, there's an analogous feature "[Groups](https://docs.gitlab.com/ee/user/group/)", which maybe is a better term than "project"?

In Github, there's a "[project](https://help.github.com/en/articles/about-project-boards)" feature, but i'm not sure if this is analogous or not.

Should we implement this at the provider interface, or provide independent, repo-specific features?

**Comment by ataylorme** [comment]
@aaronbauman GitHub projects are more like a Trello board than groups. I would say [GitHub teams](https://help.github.com/en/articles/about-teams) are closer to GitLab groups but lack the project grouping (they are only for managing permissions of multiple users).

> Should we implement this at the provider interface, or provide independent, repo-specific features?

I'll have to think about this. From the technical side doing so at the provider interface is cleaner. However, it is more user-friendly to use the terminology of each provider (e.g. have a `--project` option for BitBucket and `--group` for GitLab).

Maybe a middle-ground would be having an abstract way for providers to add addition `build:project:create` options, then add a `--project` option for BitBucket and `--group` for GitLab.

**Comment by aaronbauman** [comment]
Created this PR to address the issue:
https://github.com/pantheon-systems/terminus-build-tools-plugin/pull/294
